samples/performance/bindless/bindless.cpp

```diff
@@ -0,0 +1,925 @@
+
+		image->create_vk_image(*device);
+		Texture texture;
+		texture.n_mip_maps = uint32_t(image->get_mipmaps().size());
+		texture.image      = std::make_unique<vkb::core::Image>(*device,
+                                                           image->get_extent(),
+                                                           image->get_format(),
+                                                           VK_IMAGE_USAGE_SAMPLED_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT,
+                                                           VMA_MEMORY_USAGE_GPU_ONLY,
+                                                           VK_SAMPLE_COUNT_1_BIT,
+                                                           texture.n_mip_maps,
+                                                           1,
+                                                           VK_IMAGE_TILING_OPTIMAL,
+                                                           0);
+
+		const auto &data        = image->get_data();
+		auto        data_buffer = std::make_unique<vkb::core::Buffer>(*device, data.size() * sizeof(data[0]), VK_BUFFER_USAGE_TRANSFER_SRC_BIT, VMA_MEMORY_USAGE_CPU_TO_GPU);
+		data_buffer->update(data.data(), data.size() * sizeof(data[0]), 0);
+
+		auto &texture_cmd = device->get_command_pool().request_command_buffer();
+		texture_cmd.begin(VK_COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT, VK_NULL_HANDLE);
+
+		VkImageSubresourceRange subresource_range = {VK_IMAGE_ASPECT_COLOR_BIT, 0, 1, 0, 1};
+		subresource_range.baseMipLevel            = 0;
+		subresource_range.levelCount              = texture.n_mip_maps;
+
+		VkImageMemoryBarrier image_barrier = vkb::initializers::image_memory_barrier();
+		image_barrier.srcAccessMask        = 0;
+		image_barrier.dstAccessMask        = 0;
+		image_barrier.image                = texture.image->get_handle();
+		image_barrier.subresourceRange     = subresource_range;
+		image_barrier.oldLayout            = VK_IMAGE_LAYOUT_UNDEFINED;
+		image_barrier.newLayout            = VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL;
+
+		// The semaphore takes care of srcStageMask.
+		vkCmdPipelineBarrier(texture_cmd.get_handle(), VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT, VK_PIPELINE_STAGE_ALL_COMMANDS_BIT, 0, 0, nullptr, 0, nullptr, 1, &image_barrier);
+
+		auto              offsets              = image->get_offsets();
+		VkBufferImageCopy region               = {};
+		region.imageSubresource.aspectMask     = VK_IMAGE_ASPECT_COLOR_BIT;
+		region.imageSubresource.mipLevel       = 0;
+		region.imageSubresource.baseArrayLayer = 0;
+		region.imageSubresource.layerCount     = 1;
+		region.imageExtent                     = image->get_extent();
+		region.bufferOffset                    = offsets[0][0];
+
+		texture_cmd.copy_buffer_to_image(*data_buffer, *texture.image, {region});
+		texture_cmd.end();
+
+		auto &queue = device->get_queue_by_flags(VK_QUEUE_GRAPHICS_BIT, 0);
+		queue.submit(texture_cmd, device->request_fence());
+		device->get_fence_pool().wait();
```

Comment:
Should probably add `device->get_fence_pool().reset();` after this, else the request-pool grows in size for each mesh. This shouldn't _really_ be a problem -- would just result in a bunch of already-waited-on-fenced being redundantly passed to WaitForFences -- however, without the reset, I'm getting occasional DEVICE_LOST on my RTX2060 on Windows.